### PR TITLE
Add chat screen

### PR DESCRIPTION
**Pull request by fmzbl**
This changes depend on adding this (https://github.com/facumendizabal/makepad/commit/455c5a0a9ede6b7b03b3a82f952ed57639c351a2) to the stack_view

- Add chat screen
- Follow stack navigation pattern for MyProfile screen
- Create dummy DB/API
- Fix bug that when clicking moments the app freezes

image

### CODE REVIEW

File: src/api.rs [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        self.chats.iter().find(|m| m.id == chat_id)
+    }
+
+    pub fn get_messages_by_chat_id(&self, chat_id: u64) -> Vec<MessageEntry> {
+        self.messages
+            .iter()
+            .filter(|m| m.chat_id == chat_id)
+            .cloned()
+            .collect()
+    }
```
Comment (jv1i) [inline src/api.rs:170]: I think for benchmarking we should make this into a hash or something better performing. But I'm ok with moving forward with this now

File: src/app.rs [hunk 2]
```diff
@@ -291,8 +334,31 @@ impl AppMain for App {
...
+            match action.action() {
+                ChatListAction::Click(id) => {
+                    let db = Db::new();
+
+                    let mut stack_navigation = ui.get_stack_navigation(id!(navigation));
+                    if let Some(chat_entry) = db.get_chat(id) {
+                        stack_navigation
+                            .get_label(id!(chat_stack_view.title))
+                            // TODO: get username from duymmy db
+                            // .set_label(&id);
```
Comment (jv1i) [inline src/app.rs]: NIT: I think we can remove this comment

File: src/home/chat_list.rs [hunk 3]
```diff
@@ -1,68 +1,80 @@
...
-        // <Divider> {}
+            // TODO: implement divider when moving this into a widget.
+            // currently nested frames don't get drawn in the ListView
+            // <Divider> {}
     }
 
+    // TODO: implement a divider
+    // currently nested frames don't get drawn in the ListView
+    // <Divider> {}
+
```
Comment (jv1i) [inline src/home/chat_list.rs]: Looks like this comment got duplicated, can we remove both?

**Review by yulnr — APPROVED** [review]
Great job! 👏🏽 
Leaving some minor comments

**Pull request merged**